### ISSUE TITLE
[SonicColorsUltimate] Microsoft.NET folder duplication upon saving mods, cannot play game with mod loader - Steam Deck

### ISSUE THREAD

**Issue by RRR-94** [thesupersonic16/HedgeModManager#921]
Every time I press "Save" on the mod manager, I receive this error. For some reason, the Microsoft.NET folder gets replaced with a linked directory and a duplicate of this folder gets created as Microsoft.NET.bak. I suspect this is what is stopping me from saving mods for Colors Ultimate. 

Should also note I am using this on Steam Deck, so I'm not sure if this is Steam Deck exclusive.

[HMM_Snapshot_00092024170839.txt](https://github.com/user-attachments/files/17100661/HMM_Snapshot_00092024170839.txt)

HMM Info:
```
    Version: 7.12-9
    Args: 
    StartDir: Z:\home\deck\.local\share\Steam\steamapps\common\SonicColorsUltimate\exec
    Process Level: Administrator
    Game: Sonic Colors: Ultimate
    GameInstall: Sonic Colors: Ultimate
    Update Status: Up To Date
```

Exception:
```
    Type: IOException
    Message: Cannot create 'Z:\home\deck\.steam\steam\steamapps\compatdata\2055290\pfx\drive_c\windows\Microsoft.NET.bak' because a file or directory with the same name already exists.
    Source: mscorlib
    Function: Void Move(System.String, System.String)
    StackTrace: 
      at System.IO.Directory.Move (System.String sourceDirName, System.String destDirName) [0x000d8] in <de4d51d88e3b4eea89a8d6a3a9854818>:0 
      at HedgeModManager.Linux.LinkRuntimeToProtonPrefix (HedgeModManager.Game game) [0x000b9] in <4d6308b30a3b49dd800b94b1060b82d3>:0 
      at HedgeModManager.MainWindow.SaveConfig (System.Boolean startGame) [0x000c2] in <4d6308b30a3b49dd800b94b1060b82d3>:0 
```

**Comment by thesupersonic16** [comment]
I believe this issue has been reported before. HMM is not detecting .NET Framework and is trying to link it's own framework which appears to be Mono.
HMM normally would not even start up on Mono, so I am not exactly sure what is happening since it's meant to install dotnet48. 

Does SCU still work including its mods if you launched the game using the Steam client? if so, I may not need to check for Mono.

**Comment by RRR-94** [comment]
> I believe this issue has been reported before. HMM is not detecting .NET Framework and is trying to link it's own framework which appears to be Mono. HMM normally would not even start up on Mono, so I am not exactly sure what is happening since it's meant to install dotnet48.
> 
> Does SCU still work including its mods if you launched the game using the Steam client? if so, I may not need to check for Mono.

No. The game crashes as soon as I play the game from Steam client and sends me back. I can't play the game with the mod loader installed as it is.

As far as the dotnet48 goes, I should already have this installed in Bottles. Is there a way to skip this step seeing as I shouldn't need it?

**Comment by thesupersonic16** [comment]
The modloader needs dotnet48. I will need to check this issue out next time I use the deck. For now you can try the method below. I will be marking this issue as a bug.

- Manually installing libraries from another system.
  - Download [windows.7z](https://files.gamebanana.com/bitpit/windows.7z).
  - Locate the game's prefix. Usually located in `~/.local/share/Steam/steamapps/compatdata/<GAMEID>/pfx`
  - Enter the drive_c directory of the game prefix then `windows` and delete or rename the `Microsoft.NET` directory
  - Extract archive contents into the game's `drive_c` directory

**Comment by RRR-94** [comment]
> The modloader needs dotnet48. I will need to check this issue out next time I use the deck. For now you can try the method below. I will be marking this issue as a bug.
> 
> * Manually installing libraries from another system.
>   
>   * Download [windows.7z](https://files.gamebanana.com/bitpit/windows.7z).
>   * Locate the game's prefix. Usually located in `~/.local/share/Steam/steamapps/compatdata/<GAMEID>/pfx`
>   * Enter the drive_c directory of the game prefix then `windows` and delete or rename the `Microsoft.NET` directory
>   * Extract archive contents into the game's `drive_c` directory

Thank you! I can play the game with Re-Rainbow now :) have a good evening!

**Comment by akamax928** [comment]
> The modloader needs dotnet48. I will need to check this issue out next time I use the deck. For now you can try the method below. I will be marking this issue as a bug.
> 
> * Manually installing libraries from another system.
>   
>   * Download [windows.7z](https://files.gamebanana.com/bitpit/windows.7z).
>   * Locate the game's prefix. Usually located in `~/.local/share/Steam/steamapps/compatdata/<GAMEID>/pfx`
>   * Enter the drive_c directory of the game prefix then `windows` and delete or rename the `Microsoft.NET` directory
>   * Extract archive contents into the game's `drive_c` directory

It asks me to put a registry key in the microsoft.net folder

**Comment by thesupersonic16** [comment]
HMM 7 is nolonger being supported on Linux. Use HMM 8 beta instead, closing.
